cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaLocalizedProduct.js

```diff
@@ -548,29 +581,39 @@ function algoliaLocalizedProduct(parameters) {
     if (empty(product)) {
         this.id = null;
     } else {
-        if (parameters.isVariant) {
+        if (parameters.isVariant && jobHelper.isSimpleProduct(product)) { // set `variantID` explicitly to null in the `variants` array object for simple products
+            this.variantID = null;
+        } else if (parameters.isVariant) { // if variant, but building model for the record, not the `variants` array
             this.variantID = product.ID;
```

Comment:
The `isVariant` parameter is only used when building the `variants` array, so the 2nd comment is not accurate.
```suggestion
        if (parameters.isVariant) {
            // set `variantID` explicitly to null in the `variants` array object for simple products
            this.variantID = jobHelper.isSimpleProduct(product) ? : null : product.ID;
```